Apps/phsignalfx/signalfx_connector.py

```diff
@@ -0,0 +1,469 @@
+        resp_json = None
+
+        try:
+            request_func = getattr(requests, method)
+        except AttributeError:
+            return RetVal(
+                action_result.set_status(phantom.APP_ERROR, "Invalid method: {0}".format(method)),
+                resp_json
+            )
+
+        # Create a URL to connect to
+        url = "{0}{1}".format(self._base_url, endpoint)
+
+        try:
+            r = request_func(
+                url,
+                verify=config.get('verify_server_cert', False),
+                **kwargs
+            )
+        except requests.exceptions.InvalidSchema:
+            error_message = 'Error connecting to server. No connection adapters were found for %s' % (url)
+            return RetVal(action_result.set_status(phantom.APP_ERROR, error_message), resp_json)
+        except requests.exceptions.InvalidURL:
+            error_message = 'Error connecting to server. Invalid URL %s' % (url)
+            return RetVal(action_result.set_status(phantom.APP_ERROR, error_message), resp_json)
+        except requests.exceptions.ConnectionError:
+            error_message = 'Error Details: Connection refused from the server for URL: %s' % (url)
+            return RetVal(action_result.set_status(phantom.APP_ERROR, error_message), resp_json)
+        except Exception as e:
+            err = self._get_error_message_from_exception(e)
+            error_message = "Error Connecting to server. {0}".format(err)
+            return RetVal(action_result.set_status(phantom.APP_ERROR, error_message), resp_json)
+
+        return self._process_response(r, action_result)
+
+    def _handle_test_connectivity(self, param):
+        # Add an action result object to self (BaseConnector) to represent the action for this param
+        action_result = self.add_action_result(ActionResult(dict(param)))
+
+        self.save_progress("Connecting to endpoint")
+
+        # make rest call
+        ret_val, response = self._make_rest_call(
+            '/v2/dimension', action_result, params=None, headers=self._headers
+        )
+
+        if phantom.is_fail(ret_val):
+            self.save_progress("Test Connectivity Failed")
+            return action_result.get_status()
+
+        # Return success
+        self.save_progress("Test Connectivity Passed")
+        return action_result.set_status(phantom.APP_SUCCESS)
+
+    def _handle_run_query(self, param):
+        self.save_progress("In action handler for: {0}".format(self.get_action_identifier()))
+        action_result = self.add_action_result(ActionResult(dict(param)))
+
+        query_string = param['query']
+
+        # make rest call
+        # myNote: Zusammenbauen!!!
+
+        params = {
+            'query': query_string
+        }
+
+        ret_val, response = self._make_rest_call(
+            '/v2/dimension', action_result, params=params, headers=self._headers
+        )
+
+        if phantom.is_fail(ret_val):
+            return action_result.get_status()
+
```

Comment:
Please add the **total_results** summary.

Reply:
Fixed